apis/logging/v1/index_management_types.go

```diff
@@ -47,8 +47,33 @@ type IndexManagementPhasesSpec struct {
 // +k8s:openapi-gen=true
 type IndexManagementDeletePhaseSpec struct {
 	// The minimum age of an index before it should be deleted (e.g. 10d)
-	//
 	MinAge TimeUnit `json:"minAge"`
+
+	// How often to run a new prune-namespaces cron job
+	// +required
+	PrunePollInterval TimeUnit `json:"prunePollInterval"`
+
+	// The per namesapce specification to delete documents older than a given minimum age
+	// +optional
+	Namespaces []IndexManagementDeleteNamespaceSpec `json:"namespaceSpec,omitempty"`
+}
+
+type IndexManagementDeleteNamespaceSpec struct {
+	// The unique name of the spec
+	// +required
+	Name string `json:"name"` // delete-<namespace_name>
```

Comment:
This one is from the original spike document. I honestly do not use it anywhere inside the code. I'll ask Sashank why he had it there originally. Otherwise I believe it can be omitted